torchrl/modules/tensordict_module/ensemble.py

```diff
@@ -0,0 +1,137 @@
+from typing import Callable
+
+import torch
+from tensordict import TensorDict
+from tensordict.nn import make_functional, TensorDictModuleBase
+from torch import nn
+
+
+class Reduce(TensorDictModuleBase):
+    """A reduction operator that reduces across the ensemble dimension.
+
+    Args:
+        in_keys (list[str]): The input keys to reduce, this must be length one.
+        out_keys (list[str]): The output keys to reduce, this must be length one.
+        reduce_function (Callable): The function to use to reduce across the ensemble dimension.
+
+    Examples:
+        >>> import torch
+        >>> from torch import nn
+        >>> from tensordict.nn import TensorDictModule, TensorDictSequential
+        >>> from torchrl.modules import EnsembleModule, Reduce
+        >>> from tensordict import TensorDict
+        >>> module = TensorDictModule(nn.Linear(2, 3), in_keys=["bork"], out_keys=["dork"])
+        >>> m0 = EnsembleModule(module, num_copies=2)
+        >>> m1 = Reduce(["dork"], ["spork"])
+        >>> seq = TensorDictSequential(m0, m1)
+        >>> td = TensorDict({"bork": torch.randn(5, 2)}, batch_size=[5])
+        >>> seq(td)
+        TensorDict(
+            fields={
+                bork: Tensor(shape=torch.Size([5, 2]), device=cpu, dtype=torch.float32, is_shared=False),
+                dork: Tensor(shape=torch.Size([5, 3]), device=cpu, dtype=torch.float32, is_shared=False),
+                spork: Tensor(shape=torch.Size([5, 3]), device=cpu, dtype=torch.float32, is_shared=False)},
+            batch_size=torch.Size([5]),
+            device=None,
+            is_shared=False)
+        >>> m0(td).shape
+        torch.Size([2, 5])
+        >>> seq(td).shape
+        torch.Size([5])
+    """
+    def __init__(
+        self,
+        in_keys: list[str],
+        out_keys: list[str],
+        reduce_function: Callable[[torch.Tensor], torch.Tensor] = lambda x: x.min(dim=0).values,
```

Comment:
The best way to approach this is
```suggestion
        reduce_function: Callable[[torch.Tensor], torch.Tensor] = None,
```
then
```python
if reduce_function is None:
    reduce_function = lambda x: x.min(dim=0).values
```
But do we really want to have `"min"` as a default?

Reply:
What about Enum to select which op?
```
REDUCE_OP.min
REDUCE_OP.max
REDUCE_OP.sum
REDUCE_OP.any
REDUCE_OP.all
```
And then we can ask the user along which dim the op should be done.
That dim would also be the dim along which we do the tensordict indexing.